cvmfs/catalog_mgr_rw.h

```diff
@@ -160,8 +174,72 @@ class WritableCatalogManager : public SimpleCatalogManager {
    * Directories don't have extended attributes at this point.
    */
   XattrList empty_xattrs;
+
+  /**
+   * Balancer thresholds
+   */
+  const bool is_balanced_;
+  const unsigned max_weight_;
+  const unsigned min_weight_;
+  const unsigned balance_weight_;
 };  // class WritableCatalogManager
 
+
+template <class CatalogMgrT>
```

Comment:
The `CatalogBalancer` should come with a nice comment paragraph.  Why does it exist, what's its relationship to other classes in cvmfs, how does it roughly solve the problem.